client/components/editor/teaching-elements/Assessment/ImageMultipleChoice.vue

```diff
@@ -0,0 +1,326 @@
+            <img :src="answer || './assets/img/no-image.png'" class="image-content">
+          </div>
+          <input
+            :data-index="index"
+            type="file"
+            :disabled="disabled"
+            @change="updateAnswer"
+            class="form-control image-input"
+            :accept="imageInputType">
+          <span
+            v-for="(message, index) in getErrorMessages(index)"
+            :key="index"
+            class="error-msg">
+            {{ message }}
+          </span>
+        </span>
+        <span @click="removeAnswer(index)" class="mdi mdi-close control"></span>
+      </li>
+    </ul>
+  </div>
+</template>
+
+<script>
+import cloneDeep from 'lodash/cloneDeep';
+import range from 'lodash/range';
+import { blobToDataURL } from 'blob-util';
+
+const maxImageDimension = 256;
+
+const formAlert = (message) => {
+  return {
+    type: 'alert-danger',
+    text: message
+  };
+};
+
+const getImageDimensions = function (imageDataUrl) {
+  return new Promise(resolve => {
+    let i = document.createElement('img');
+    i.onload = () => resolve({ width: i.width, height: i.height });
+    i.src = imageDataUrl;
+  });
+};
+
+class ImageValidationError extends Error {
+  constructor(message) {
+    super(message);
+    this.name = 'ImageValidationError';
+  }
+}
+
+export default {
+  props: {
+    assessment: Object,
+    isEditing: Boolean,
+    errors: Array,
+    fileInputTypes: { type: Array, default: () => ['image/jpeg', 'image/png'] }
+  },
+  data() {
+    return {
+      fieldErrors: []
+    };
+  },
+  computed: {
+    answers() {
+      return this.assessment.answers;
+    },
+    correct() {
+      return this.assessment.correct;
+    },
+    feedback() {
+      return this.assessment.feedback;
+    },
+    disabled() {
+      return !this.isEditing;
+    },
+    imageInputType() {
+      return this.fileInputTypes.join(',');
+    }
+  },
+  methods: {
+    update(data) {
+      this.$emit('update', data);
+    },
+    toggleAnswer(index) {
+      let correct = cloneDeep(this.correct);
+      const position = correct.indexOf(index);
+
+      if (position < 0) {
+        correct.push(index);
+      } else {
+        correct.splice(position, 1);
+      }
+
+      this.update({ correct });
+    },
+    updateAnswer(evt) {
+      let answers = cloneDeep(this.answers);
+      const { target: input } = evt;
+      const { dataset, files = [] } = input;
+      const imageFile = files[0];
+
+      return new Promise((resolve, reject) => {
+        this.resetErrorMessages(dataset.index);
+        if (imageFile.size > 100000) {
+          return reject(new ImageValidationError(
+            'The image is too large. Maximum allowed image size: 100 kB'
+          ));
+        }
+        return resolve(imageFile);
+      })
+        .then(() => blobToDataURL(imageFile))
+        .then(imageDataUrl => {
+          answers[dataset.index] = imageDataUrl;
+          return getImageDimensions(imageDataUrl);
```

Comment:
You should modify answers array only if image passes validations ❕ 
So please create a helper function that will receive image and do all required validations :bowtie: